src/components/views/context_menus/MessageContextMenu.js

```diff
@@ -316,77 +329,96 @@ export default class MessageContextMenu extends React.Component {
             isUrlPermitted(mxEvent.event.content.external_url)
         ) {
             externalURLButton = (
-                <MenuItem
+                <IconizedContextMenuOption
+                    iconClassName="mx_MessageContextMenu_iconLink"
+                    onClick={this.closeMenu}
+                    label={ _t('Source URL') }
                     element="a"
-                    className="mx_MessageContextMenu_field"
                     target="_blank"
                     rel="noreferrer noopener"
-                    onClick={this.closeMenu}
                     href={mxEvent.event.content.external_url}
-                >
-                    { _t('Source URL') }
-                </MenuItem>
+                />
             );
         }
 
         if (this.props.collapseReplyThread) {
             collapseReplyThread = (
-                <MenuItem className="mx_MessageContextMenu_field" onClick={this.onCollapseReplyThreadClick}>
-                    { _t('Collapse Reply Thread') }
-                </MenuItem>
+                <IconizedContextMenuOption
+                    iconClassName="mx_MessageContextMenu_iconCollapse"
+                    label={_t("Collapse reply thread")}
+                    onClick={this.onCollapseReplyThreadClick}
+                />
             );
         }
 
         let reportEventButton;
         if (mxEvent.getSender() !== me) {
             reportEventButton = (
-                <MenuItem className="mx_MessageContextMenu_field" onClick={this.onReportEventClick}>
-                    { _t('Report Content') }
-                </MenuItem>
+                <IconizedContextMenuOption
+                    iconClassName="mx_MessageContextMenu_iconReport"
+                    label={_t("Report")}
+                    onClick={this.onReportEventClick}
+                />
             );
         }
 
-        return (
-            <div className="mx_MessageContextMenu">
-                { resendReactionsButton }
-                { redactButton }
+        optionLists.push((
+            <IconizedContextMenuOptionList key={'group1'}>
+                { quoteButton }
                 { forwardButton }
                 { pinButton }
-                { viewSourceButton }
-                { unhidePreviewButton }
                 { permalinkButton }
-                { quoteButton }
+                { reportEventButton }
                 { externalURLButton }
+                { unhidePreviewButton }
+                { viewSourceButton }
+                { resendReactionsButton }
                 { collapseReplyThread }
-                { reportEventButton }
-            </div>
+            </IconizedContextMenuOptionList>
+        ));
+
+        if (redactButton) {
+            optionLists.push((
+                <IconizedContextMenuOptionList key={'group2'} red>
+                    { redactButton }
+                </IconizedContextMenuOptionList>
+            ));
+        }
```

Comment:
using an array here does not match our typical style for conditional rendering.

Just like with the buttons themselves, I suggest just having a null variable which you conditionally set

Reply:
✓ done